src/hooks/useCheckImage.ts

```diff
@@ -1,22 +1,41 @@
+import { useMemo } from "react";
 import { useAsync } from "react-use";
 
 export const useCheckImage = (
-  src: string
+  src: string | undefined
 ): {
   isValid: boolean;
   isLoading: boolean;
   error: string | undefined;
 } => {
   const { loading, error } = useAsync(async () => {
     const checkImage = new Image();
+    if (!src) throw Error("src is not defined.");
     checkImage.src = src;
 
     await checkImage.decode();
   }, [src]);
 
-  return {
-    isValid: !error,
-    isLoading: loading,
-    error: error?.message,
-  };
+  return useMemo(
+    () => ({
+      //here we consider the image valid event while it still loading
```

Comment:
typo?
```suggestion
      // here we consider the image valid even while it's still loading
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2025/commits/5f89f4dfa98705fd30dad433d7627dd14c1c9057